src/EA.Weee.Web/Areas/Producer/Controllers/ProducerSubmissionController.cs

```diff
@@ -354,16 +359,39 @@ public async Task<ActionResult> EditContactDetails(EditContactDetailsViewModel m
         [SmallProducerSubmissionContext]
         public async Task<ActionResult> AppropriateSignatory()
         {
-           return View();
+            var model = mapper.Map<SmallProducerSubmissionData, AppropriateSignatoryViewModel>(SmallProducerSubmissionData);
+
+            await SetBreadcrumb(SmallProducerSubmissionData.OrganisationData.Id, ProducerSubmissionConstant.NewContinueProducerRegistrationSubmission);
+
+            return View();
         }
 
         [HttpPost]
         [ValidateAntiForgeryToken]
         [SmallProducerSubmissionContext]
-        public async Task<ActionResult> AppropriateSignatory(object model) // needs to be updated to the final model.
+        public async Task<ActionResult> AppropriateSignatory(AppropriateSignatoryViewModel model)
+        {
+            if (ModelState.IsValid)
+            {
+                var request = addSignatoryAndCompleteRequestCreator.ViewModelToRequest(model);
+
+                using (var client = apiClient())
+                {
+                    await client.SendAsync(User.GetAccessToken(), request);
+                }
+
+                return await RedirectToNextUrl();
+            }
+
+            await SetBreadcrumb(model.OrganisationId, ProducerSubmissionConstant.NewContinueProducerRegistrationSubmission);
+
+            return View(model);
+        }
+
+        private async Task<ActionResult> RedirectToNextUrl()
         {
             var existingPaymentInProgress = await paymentService.CheckInProgressPaymentAsync(User.GetAccessToken(),
```

Comment:
@stevelee-civica  Dont worry about creating unit tests for the payment process I will do that as part of my changes